lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx

```diff
@@ -0,0 +1,219 @@
+                                                }}
+                                            >
+                                                <EditIcon />
+                                            </button>
+                                            <button
+                                                className='btn'
+                                                onClick={() => handleDeleteMetadata(metadata, index)}
+                                            >
+                                                <DeleteIcon />
+                                            </button>
+                                        </div>
+
+                                        <MetadataEditModal index={index} metadata={metadata} modal={modal} />
+                                    </td>
+                                </tr>
+                            );
+                        })}
+                    </tbody>
+                </table>
+                <div className={'flex justify-end'}>
+                    <button
+                        className='btn'
+                        onClick={() => {
+                            addNewMetadata();
+                        }}
+                    >
+                        <AddIcon />
+                    </button>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+export function MetadataEditModal({ index, metadata, modal }: { index: number; metadata: Metadata; modal: string }) {
+    const { updateMetadata } = useContext(ConfigContext);
+
+    const [metadataType, setMetadataType] = useState<MetadataType>(metadata.type);
+    const [metadataName, setMetadataName] = useState(metadata.name);
+    const [generateIndex, setGenerateIndex] = useState(metadata.generateIndex);
+
+    const handleUpdateMetadata = () => {
+        updateMetadata({ ...metadata, name: metadataName, type: metadataType, generateIndex }, index);
+    };
+
+    return (
+        <dialog id={modal} className='modal'>
+            <div className='modal-box'>
+                <h3 className='font-bold text-lg'>Edit metadata</h3>
+                <div className='flex flex-col space-x-2 items-center'>
+                    <table className='table'>
+                        <tbody>
+                            <tr>
+                                <td>Name</td>
+                                <td>
+                                    <EditMetadataName metadataName={metadataName} setMetadataName={setMetadataName} />
+                                </td>
+                            </tr>
+
+                            <tr>
+                                <td>Type</td>
+                                <td>
+                                    <SelectMetadataType metadataType={metadataType} setMetadataType={setMetadataType} />
+                                </td>
+                            </tr>
+
+                            <tr>
+                                <td>Generate Index</td>
```

Comment:
Generating an index should only be possible for string columns (and for pango lineage columns it's always true).